### ISSUE TITLE
Loss function of Discriminator

### ISSUE THREAD

**Issue by Ikhwansong** [ErinChen1/EPDN#3]
Hi , Your research is wonderful for me. because I have researched same theme.
I have a question from the paper.

In fourth page, feature matching loss function equation (6) is described that which is sum of each output which is pixel-wise L1 norm result from discriminator's layer.

So, you mean the discriminator has some layers.  then If you have two outputs(dehazed image, ground truth) , you get some outputs(2 * number of layer in Discriminator) which are extracted from each image feature maps.

Finally, If your discriminator has 7-layer, this may be output that is  7 * (feature map(extracted from dehazed img) - feature map(extracted from GT)) equation.

This is I understand right?

**Comment by dcynice** [comment]
Have you implemented the code for this article? There seems to be a lack of training code

**Comment by Ikhwansong** [comment]
> Have you implemented the code for this article? There seems to be a lack of training code
Yes. There is no the code for training.
So, If you want to train, you have to implement training code with following this article.

**Comment by dcynice** [comment]
Thank you. How do you train? Can you guide me in the specific process

**Comment by 123ERFW** [comment]
Excuse me, I want to ask you a question. IndexError:list index out of range, How to solve this problem，Thanks.

**Comment by tongchangD** [comment]
@123ERFW  out of range 明显是index 超出了原本list的长度范围应该比较好解决
